woocommerce/woocommerce-ios
# Product sharing: Update UI for the AI sheet

Part of #9867 


## Description

This PR updates the UI of the product sharing sheet with AI following the recent update [pe5sF9-1AH-p2#comment-2283]. Changes include:
- Updated the generate button style to `SecondaryLoadingButtonStyle` with a title for the loading state.
- Added a small button for showing the legal page.
- To better adapt with different text sizes, the two buttons are put together in an `AdaptiveStack`. The generate button is kept with a fixed size for non-accessibility font sizes in portrait mode. 
- Replaced `sizeCategory` with `dynamicTypeSize` as it's [deprecated](https://developer.apple.com/documentation/swiftui/environmentvalues/sizecategory) soon.
- Use `localizedMessage` for error to make it clearer and better match the behavior of product description generation.

## Testing instructions

- Log in to a public WPCom store with at least 1 published product.
- Tap to share the product, notice that the AI sheet matches the design in the aforementioned comment.
- Tap the (i) button, notice that a web view is presented showing the legal page for AI content.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/4d6b4fe9-c714-469d-9bd3-cfae1d0b5dd2



---
